integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java

```diff
@@ -208,6 +208,31 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
         System.out.println("Launching task " + taskId.getValue() + " using offer "
             + offer.getId().getValue());
 
+        Protos.Resource.Builder portsBuilder = Protos.Resource.newBuilder()
+            .setName(Constants.MESOS_RESOURCE_PORTS)
+            .setType(Protos.Value.Type.RANGES);
+
+        if (executorBuilder.getSource().equals("worker")) {
+          portsBuilder.setRanges(Protos.Value.Ranges.newBuilder()
+              .addRange(Protos.Value.Range.newBuilder()
+                  .setBegin(sConf.getLong(Constants.WORKER_RPC_PORT))
+                  .setEnd(sConf.getLong(Constants.WORKER_RPC_PORT)))
+              .addRange((Protos.Value.Range.newBuilder()
+                  .setBegin(sConf.getLong(Constants.WORKER_DATA_PORT))
+                  .setEnd(sConf.getLong(Constants.WORKER_DATA_PORT))))
+              .addRange((Protos.Value.Range.newBuilder()
+                  .setBegin(sConf.getLong(Constants.WORKER_WEB_PORT))
+                  .setEnd(sConf.getLong(Constants.WORKER_WEB_PORT)))));
+        } else {
```

Comment:
instead of relying the source != "worker" implies source == "master" make this a separate `if (source == MASTER) { ... }` statement where MASTER is the named constant for "master" used here and on lines 133 and 134